### PR TITLE
Turn snippet preview title red when the length score is lower than 5

### PR DESCRIPTION
**Pull request by IreneStr**
## Summary

This PR can be summarized in the following changelog entry:

* Change the text color of the title in the snippet to red when the title is too long.

## Relevant technical choices:

* Rename descriptionLengthAssessment to descriptionLengthProgress because the object shows progress, not an assessment.
* Rename DraftJS to Draft.js

## Test instructions

This PR can be tested by following these steps:

* See https://github.com/Yoast/wordpress-seo/pull/9646

Connect https://github.com/Yoast/wordpress-seo/issues/9588

### CODE REVIEW

**Comment by IreneStr** [comment]
As discussed with @atimmer, the diff-coverage test failure is fine, and can be ignored.

File: composites/Plugin/SnippetEditor/components/ReplacementVariableEditor.js [hunk 1]
```diff
@@ -265,11 +266,13 @@ ReplacementVariableEditor.propTypes = {
 	onChange: PropTypes.func.isRequired,
 	onFocus: PropTypes.func,
 	onBlur: PropTypes.func,
+	className: PropTypes.string,
```
Comment (boblinthorst) [inline composites/Plugin/SnippetEditor/components/ReplacementVariableEditor.js]: where is this used?

File: composites/Plugin/SnippetEditor/components/SnippetEditorFields.js [hunk 2]
```diff
@@ -232,13 +233,14 @@ class SnippetEditorFields extends React.Component {
 							replacementVariables={ replacementVariables }
 							ref={ ( ref ) => this.setRef( "title", ref ) }
 							ariaLabelledBy={ this.uniqueId + "-title" }
+							titleLengthProgress={ titleLengthProgress }
```
Comment (boblinthorst) [inline composites/Plugin/SnippetEditor/components/SnippetEditorFields.js]: prop isnt caught by the replacementvareditor?

File: composites/Plugin/SnippetEditor/components/SnippetEditor.js [hunk 3]
```diff
@@ -37,6 +40,39 @@ const CloseEditorButton = SnippetEditorButton.extend`
 	margin-left: 20px;
 `;
 
+/**
+ * Gets the title progress.
```
Comment (boblinthorst) [inline composites/Plugin/SnippetEditor/components/SnippetEditor.js]: low prio: add a br between title and params.

File: composites/Plugin/SnippetEditor/components/SnippetEditor.js [hunk 4]
```diff
@@ -56,9 +92,9 @@ class SnippetEditor extends React.Component {
 	 *                                                   be in.
 	 * @param {Function} props.onChange                  Called when the data
 	 *                                                   changes.
-	 * @param {Object} props.titleLengthAssessment       The values for the title
+	 * @param {Object} props.titleLengthProgress       The values for the title
```
Comment (boblinthorst) [inline composites/Plugin/SnippetEditor/components/SnippetEditor.js]: title width assessment*

File: composites/Plugin/SnippetEditor/components/SnippetEditor.js [hunk 5]
```diff
@@ -85,6 +124,47 @@ class SnippetEditor extends React.Component {
...
+				descriptionLengthProgress: getDescriptionProgress( nextProps.data.description.length ),
+			}
+		);
+	}
+
+	/**
+	 * Handles the onChange event.
+	 *
+	 * First updates the description progress and title progress.
+	 * Then calls the onChange function that is paased through the props.
```
Comment (boblinthorst) [inline composites/Plugin/SnippetEditor/components/SnippetEditor.js]: passed*

File: composites/Plugin/SnippetEditor/components/SnippetEditor.js [hunk 6]
```diff
@@ -85,6 +124,47 @@ class SnippetEditor extends React.Component {
 		this.setEditButtonRef = this.setEditButtonRef.bind( this );
 	}
 
+	/**
+	 * Updates the state when the component receives new props.
+	 *
+	 * @param {Object} nextProps The new props
```
Comment (boblinthorst) [inline composites/Plugin/SnippetEditor/components/SnippetEditor.js]: dot

**Comment by Xyfi** [comment]
Acceptance :+1: Waiting for yoast components 1.36 release.

**Comment by IreneStr** [comment]
I've updated the branch with the newest version of YoastSEO.js

**Pull request merged**